src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+    }
+
+    @Test
+    void buildSetDescriptionUpdateAction_WithJustOldDescriptionHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getDescription()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final LocalizedString newDesc = LocalizedString.of(Locale.ENGLISH, "new-desc");
+        when(newCartDiscountDraft.getDescription()).thenReturn(newDesc);
+
+        final Optional<UpdateAction<CartDiscount>> setDescriptionUpdateAction =
+                buildSetDescriptionUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(setDescriptionUpdateAction).contains(SetDescription.of(newDesc));
+    }
+
+    @Test
+    void buildSetDescriptionUpdateAction_WithBothDescriptionHaveNullValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getDescription()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getDescription()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> setDescriptionUpdateAction =
+                buildSetDescriptionUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(setDescriptionUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn("0.1");
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final String sortOrder = "0.3";
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(sortOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).contains(ChangeSortOrder.of(sortOrder));
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final String sortOrder = "0.1";
+
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn(sortOrder);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(sortOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithJustNewSortOrderHasNullValue_ShouldBuildUpdateAction() {
```

Comment:
`WithOnlyNullNewSortOrder`

Reply:
what about empty string value?

Reply:
but it seems that it is also not optional from docs: https://docs.commercetools.com/http-api-projects-cartDiscounts#cartdiscountdraft